src/libraries/System.Globalization/tests/CompareInfo/CompareInfoTests.Compare.cs

```diff
@@ -44,6 +44,7 @@ public static IEnumerable<object[]> Compare_TestData()
             // PlatformDetection.IsHybridGlobalizationOnBrowser does not support IgnoreWidth
             CompareOptions ignoredOptions = PlatformDetection.IsHybridGlobalizationOnBrowser ?
                 CompareOptions.IgnoreKanaType | CompareOptions.IgnoreCase :
+                PlatformDetection.IsHybridGlobalizationOnOSX ? CompareOptions.IgnoreWidth | CompareOptions.IgnoreCase :
                 CompareOptions.IgnoreKanaType | CompareOptions.IgnoreWidth | CompareOptions.IgnoreCase;
```

Comment:
`ignoredOptions` is becoming vague and made me wonder why we aren't including `IgnoreNonSpace` for example. It seems like it originally was `ignoreKanaIgnoreWidthIgnoreCase`.

I'm wondering if all of the test cases that use this option actually need all `IgnoreKana` `IgnoreWidth` `IgnoreCase`, or if they are actually just `IgnoreKana`+`IgnoreCase` or `IgnoreWidth`+`IgnoreCase`

Its a little confusing to me that all of the test cases using this option should still pass if we are no longer ignoring kana or no longer ignoring width on different platforms.

For example, I dont quite understand if `IsHybridGlobalizationOnBrowser` doesn't support `IgnoreWidth`, then shouldn't the expected value of the first two test cases differ?
```
yield return new object[] { s_invariantCompare, "\u3042", "\u30A2", ignoredOptions, 0 };
yield return new object[] { s_invariantCompare, "\u3042", "\uFF71", ignoredOptions, 0 };
```
Where `\u3042` is Hiragana Letter A, `\u30A2` is Katakana Letter A, and `\uFF71` is Halfwidth Katakana Letter A. So with `IgnoreKana` and `IgnoreWidth`, `\u3042 == \u30A2 == \uFF71` makes sense to me
What I dont expect, is that a platform not supporting `IgnoreKanaType` would still use these test cases.

I would expect that a platform not supporting width ignoring would say that `\u30A2` and `\uFF71` are different.

Reply:
> For example, I dont quite understand if `IsHybridGlobalizationOnBrowser` doesn't support `IgnoreWidth`, then shouldn't the expected value of the first two test cases differ?

You are right, `IsHybridGlobalizationOnBrowser` does not support `IgnoreWidth` and it throws `PNSE` for it. Doing as you say, we would exclude testing fully a lot of test cases and test only the scenario: if the comparison option is passed correctly and is ignoring width - then check if early throw worked.
Instead, we prefer to use the existing cases (that actually are not impacted much by the presence of `IgnoreWidth`) and check if comparison result is fine.
I understand that the itf can be confisung and the only right solution would be to correct original tests - only a few are in fact impacted by all 3 options, they should not have all 3 of them passed every time, only the ones that really matter. I did not intend to change any existing tests, though to make sure we are not messing up the original behavior. Do you think we should do it?

Reply:
I agree, I think modifying multiple tests to check if there is an early throw for unsupported options would be redundant and clutter our tests. 

I am curious about exactly what the test cases are testing (for the tests that use a multitude of options but seemingly only need subset). Perhaps the strings really need all of the options for the test to pass on some platforms? Maybe its actually checking that passing extra options would not change the outcome? Or maybe its just a blanket combination of comparison options to cover a multitude of tests and we should fine grain them as we learned some platforms don't actually support certain comparison options.

I think we should check that reducing the comparison options to just what is needed doesn't cause any test failures on current platforms, and modify the name of the `CompareOptions` variable to be more test case specific. Maybe we can have a `ignoreKanaIgnoreWidthIgnoreCase` for tests that actually need all three and a `ignoreKanaIgnoreCase` for those that don't actually test width, etc.

Reply:
I made an attempt to clean up the original tests, we'll see what CI says https://github.com/dotnet/runtime/pull/86305.